extensions/vai/vai.go

```diff
@@ -0,0 +1,157 @@
+package vai
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/features"
+	"github.com/rancher/shepherd/extensions/settings"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"k8s.io/apimachinery/pkg/types"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	cattleSystemNamespace    = "cattle-system"
+	cattleAgentDeployment    = "cattle-cluster-agent"
+	clusterRegisterContainer = "cluster-register"
+	uiSQLCacheResource       = "ui-sql-cache"
+	cattleFeaturesEnvVar     = "CATTLE_FEATURES"
+)
+
+func updateSQLCache(adminClient *rancher.Client, value bool) error {
+	managementClient := adminClient.Steve.SteveType("management.cattle.io.feature")
+
+	steveCacheFlagResp, err := managementClient.ByID(uiSQLCacheResource)
+	if err != nil {
+		return err
+	}
+
+	featureFlagResp, err := features.UpdateFeatureFlag(adminClient.Steve, steveCacheFlagResp, value)
+	if err != nil {
+		return err
+	}
+
+	errors := pods.StatusPods(adminClient, "local")
+	if len(errors) > 1 {
+		return fmt.Errorf("error when restarting pods")
+	}
+
+	adminClient.Session.RegisterCleanupFunc(func() error {
+		_, err := features.UpdateFeatureFlag(adminClient.Steve, featureFlagResp, false)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
```

Comment:
Is this from the dev stage, it seems like the first iteration of this: https://github.com/rancher/shepherd/pull/218/files#diff-92566b04396c1f7c9b8e22aca23df2d6ffdc48fd4630c1115fbab240f2b1bcbaR87-R93

Reply:
Removed it and tested it out.